common: bwl: return error on consecutive nl-msg-get failures

adam1985d opened the pull request:
Added failure on attempt counter for dwpal_driver_nl_msg_get operation
to prevent unnecessary monitor thread restarts in case of a one-time
buffer glitch in dwpal-nl-parser.

The above-mentioned scenario can happen when 2 sockets are using
dwpal_driver_nl_msg_get simultaneously and receiving an event at the
same time.
If max-retries is reached, the process_nl_events() fails and the
monitor_thread_stop will be performed.

Fixes #1001 

Signed-off-by: Adam Dov <adamx.dov@intel.com>

kantera800 reviewed (approved):
looks good.
but please add a comment in the code that explains why the retry mechanism is needed.

adam1985d commented:
> looks good.
> but please add a comment in the code that explains why the retry mechanism is needed.

done

adam1985d committed:
common: bwl: return error on consecutive nl-msg-get failures

Added failure on attempt counter for dwpal_driver_nl_msg_get operation
to prevent unnecessary monitor thread restarts in case of a one-time
buffer glitch in dwpal-nl-parser.

The above-mentioned scenario can happen when 2 sockets are using
dwpal_driver_nl_msg_get simultaneously and receiving an event at the
same time.
If max-retries is reached, the process_nl_events() fails and the
monitor_thread_stop will be performed.

Signed-off-by: Adam Dov <adamx.dov@intel.com>
